src/cursor/cur_file.c

```diff
@@ -693,6 +693,10 @@ __curfile_create(WT_SESSION_IMPL *session,
 		cacheable = false;
 	}
 
+	WT_ERR(__wt_config_gets_def(session, cfg, "read_once", 0, &cval));
+	if (cval.val != 0)
+		F_SET(cbt, WT_CBT_READ_ONCE);
```

Comment:
I think we either have to set `cacheable = false` here and add a `read_once` case to https://github.com/wiredtiger/wiredtiger/blob/develop/src/cursor/cur_std.c#L731-L761, or deal with setting / clearing `WT_CBT_READ_ONCE` in this code: https://github.com/wiredtiger/wiredtiger/blob/develop/src/cursor/cur_std.c#L801-L822

I'd prefer the latter, so that we can use the cursor cache with `read_once` set.

Reply:
I got #2 to work. That, in combination with @sueloverso's testing suggestion, I could observe the caching logic from being broken (not clearing the flag) to working. Two remarks:
1) Should we spend time coming up with a more preferable way to detect cursor caching bugs regarding `read_once`? A failing test result is certainly subtle to diagnose.
2) I looked at the first character of the uri + internal_uri to determine if the `WT_CURSOR` object was really a `WT_CURSOR_BTREE` object, but I'm not sure if I can just assume `internal_uri` is always available when pulling from the cache and the right thing to check?